### ISSUE TITLE
[BUG] Setting a Series value with boolean index fails

### ISSUE THREAD

**Issue by randerzander** [rapidsai/cudf#3049]
This works:
```
import pandas as pd

df = pd.DataFrame()
df['val'] = [0.1, 0.2, 0.3]

df[df['val'] > 0.2] = 1
```

The cuDF version fails:
```
import cudf

df = cudf.DataFrame()
df['val'] = [0.1, 0.2 , 0.3]

df[df['val'] > 0.2] = 1
```
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-6b3b8a75f210> in <module>
----> 1 df[df['val'] > 0.2] = 1

/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/dataframe.py in __setitem__(self, name, col)
    379                 self._cols[col_name][mask] = col
    380 
--> 381         elif name in self._cols:
    382             self._cols[name] = self._prepare_series_for_add(col)
    383         else:

TypeError: unhashable type: 'Series'
```

**Comment by VibhuJawa** [comment]
A couple of clarifying questions: 

In the example you are setting the whole`dataframe` using the `bool-index` but the issue says you want to set series value with `boolean` `index` .

 Can you clarify which is the behavior you want to support. 

**For clarification:** 

Below works:
```
import cudf

df = cudf.DataFrame()
df['val'] = [0.1, 0.2, 0.3]

df['val'][df['val'] > 0.1] = 1
print(df)           
```
And this fails: 
```
import cudf

df = cudf.DataFrame()
df['val'] = [0.1, 0.2 , 0.3]

df[df['val'] > 0.2] = 1
```

**Comment by beckernick** [comment]
@VibhuJawa @randerzander , you may want to coordinate with @thomcom , who picked up https://github.com/rapidsai/cudf/issues/2758 . I suspect the resolution to 2758 will resolve this issue as well.

**Comment by thomcom** [comment]
Indeed, this is actually a duplicate of #2758 since `df['val'][df['val'] > 0.1] = 1` works. @VibhuJawa, do you mind picking up #2758? It should be pretty easy - just iterate over the columns and do it.

**Comment by VibhuJawa** [comment]
@thomcom , Sure will pick it up. 
Can you assign it to me so that i can track it easily.

**Comment by beckernick** [comment]
@VibhuJawa , assigning #2758 it to you now and removing @thomcom

**Comment by kkraus14** [comment]
Fixed by #3442
